bot-components/GitHub_queries.ml

```diff
@@ -267,6 +267,110 @@ let get_pull_request_refs ~bot_info ~owner ~repo ~number =
           f "Query pull_request_info failed with %s" err)
   >|= Result.bind ~f:(pull_request_info_of_resp ~owner ~repo ~number)
 
+let pull_request_reviews_info_of_resp ~owner ~repo ~number resp :
+    (GitHub_subscriptions.pull_request_reviews_info, string) Result.t =
+  match resp#repository with
+  | None ->
+      Error (f "Unknown repository %s/%s." owner repo)
+  | Some repository -> (
+    match repository#pullRequest with
+    | None ->
+        Error (f "Unknown pull request %s/%s#%d." owner repo number)
+    | Some pull_request -> (
+      match
+        ( pull_request#baseRef
+        , pull_request#files
+        , pull_request#approvedReviews
+        , pull_request#commentReviews
+        , pull_request#reviewDecision )
```

Comment:
Since in the end we do not error if `reviewDecision` is `None`, you do not actually need to match on it at this point.